src/lib/profiles/data-management/Current/SubscriptionHandler.cpp

```diff
@@ -408,6 +408,12 @@ WEAVE_ERROR SubscriptionHandler::ParsePathVersionEventLists(SubscribeRequest::Pa
             else
             {
                 uint64_t existingVersion;
+
+#if WEAVE_CONFIG_PERSIST_TRAIT
+                err = platform::UpdateVersionFromPersistedStorage(dataSource);
```

Comment:
For this, we are updating the version for only those in the SubscribeRequest. I guess we can go over all the sources in the Catalog before a subscription is attempted and update the versions then? That way we do not need to insert a version update logic here, potentially.